src/test/test_data/go_test_failure.txt

```diff
@@ -1,9 +1,9 @@
 === RUN TestJSONExpectedFailure-6
---- FAIL: TestJSONExpectedFailure-6 (0.00s)
 	results_test.go:11: Unable to parse file: EOF
+--- FAIL: TestJSONExpectedFailure-6 (0.00s)
```

Comment:
I'm not fully certain about all the logic here but I don't observe this ordering happening to me when tests fail.